clients/vault/src/stellar_oracle/mod.rs

```diff
@@ -0,0 +1,211 @@
+            .get_vec()
+            .into_iter()
+            .filter_map(|op| match &op.body {
+                stellar_relay::sdk::types::OperationBody::Payment(p) => {
+                    let d = p.destination.clone();
+                    if VAULT_ADDRESSES_FILTER.contains(&std::str::from_utf8(d.to_encoding().as_slice()).unwrap()) {
+                        Some(p)
+                    } else {
+                        None
+                    }
+                }
+                _ => None,
+            })
+            .collect();
+
+        if payment_ops_to_vault_address.len() == 0 {
+            // The transaction is not relevant to use since it doesn't
+            // include a payment to our vault address
+            return false;
+        } else {
+            tracing::info!("Transaction to our vault address received.");
+            let source = transaction.source_account.clone();
+            for payment_op in payment_ops_to_vault_address {
+                let destination = payment_op.destination.clone();
+                let amount = payment_op.amount;
+                let asset = payment_op.asset.clone();
+                tracing::info!("Deposit amount {:?} stroops", amount);
+                // print_asset(asset);
+                tracing::info!(
+                    "From {:#?}",
+                    std::str::from_utf8(source.to_encoding().as_slice()).unwrap()
+                );
+                tracing::info!(
+                    "To {:?}",
+                    std::str::from_utf8(destination.to_encoding().as_slice()).unwrap()
+                );
+            }
+            return true;
+        }
+    }
+}
+
+fn print_asset(asset: Asset) {
+    match asset {
+        Asset::AssetTypeNative => tracing::info!("XLM"),
+        Asset::AssetTypeCreditAlphanum4(value) => {
+            tracing::info!("{:?}", std::str::from_utf8(value.asset_code.as_slice()).unwrap());
+            tracing::info!(
+                "{:?}",
+                std::str::from_utf8(value.issuer.to_encoding().as_slice()).unwrap()
+            );
+        }
+        Asset::AssetTypeCreditAlphanum12(value) => {
+            tracing::info!("{:?}", std::str::from_utf8(value.asset_code.as_slice()).unwrap());
+            tracing::info!(
+                "{:?}",
+                std::str::from_utf8(value.issuer.to_encoding().as_slice()).unwrap()
+            );
+        }
+        Asset::Default(code) => tracing::info!("Asset type {:?}", code),
+    }
+}
+
+/// Returns an ActorHandler that will allow us to get the map.
+/// ```
+/// loop {
+/// 		tokio::time::sleep(Duration::from_secs(8)).await;
+///
+/// 		let (sender,receiver) = tokio::sync::oneshot::channel();
+/// 		handler.get_map(sender).await?;
+///
+/// 		if let Ok(res) = receiver.await {
+/// 			log::info!("the map {:?}", res.envelopes_map().len());
+/// 		}
+/// 	}
+/// ```
+pub async fn handler(node_info: NodeInfo, connection_cfg: ConnConfig) -> Result<ActorHandler, Error> {
```

Comment:
This right here will run similarly with the demo, and returns a handler that  we can send a message to, to retrieve the map.

Reply:
but i'm thinking, that it doesn't have to be the map. It could just be proof.